signal_timeout_block.py

```diff
@@ -63,13 +67,15 @@ def _cancel_timeout_jobs(self, key):
         for job in self._jobs[key].values():
             job.cancel()
 
-    def _schedule_timeout_job(self, key, interval, repeatable):
+    def _schedule_timeout_job(self, signal, key, interval, repeatable):
         self._logger.debug("Scheduling new timeout job for group {}, interval"
                            "={} repeatable={}".format(
                                key, interval, repeatable))
         self._jobs[key][interval] = Job(
-            self._timeout_job, interval, repeatable, key, interval)
+            self._timeout_job, interval, repeatable, signal, key, interval)
 
-    def _timeout_job(self, key, interval):
+    def _timeout_job(self, signal, key, interval):
         """ Triggered when an interval times out """
-        self.notify_signals([Signal({'timeout': interval, 'group': key})])
+        signal.timeout = interval
```

Comment:
We do tend to use setattr but this appears to be working in the unittests

```
If the variable (attribute) name is known at "compile" time (hard-coded):
   e.foo = 1
If the variable name is determined at runtime:
   setattr(e, name, value)
```
source: http://rosettacode.org/wiki/Add_a_variable_to_a_class_instance_at_runtime#Python